### ISSUE TITLE
Group Discussion

### ISSUE THREAD

**Issue by bkinney** [mcjelewis/threadz#1]
Mat,

Thank you for this. I was able to install it and plug it into my own API infrastructure without undue difficulty. No sooner did I have it running than somebody wants a new feature. Of course. The issue is with group discussions. Posts created within the context of a group do not appear in the data. In fact, there is a completely different endpoint for  the /view when you are looking for group posts. I'm thinking of adding in some code to loop through all the groups to find same-named discussions. Unfortunately, the discussion id is not consistent. 

It's not clear to me that having a bunch of small networks would be particularly interesting to users, and I'm aware that students who hit the tool won't have access to all the data, but that's probably for the best anyway. 

Becky Kinney

**Comment by mcjelewis** [comment]
Yes, groups are a little bit of a different beast.  In the latest update I've added into the topic data array a couple data points that can help move this into future features.  The topic data now collects from the api call 'topic_children' and 'root_topic_id'. As you noted, to collect the discussion data from a group discussion a different api call needs to be made since the data is not a part of the course, but instead the group. With the data coming out of the 'topic_children' we could loop through the group discussion and associated them back with the original course discussion (aka 'root_topic_id').  

I'll add this functionality to future development.

thanks.
